business-registry-dashboard/app/components/Table/AffiliatedEntity/Action.vue

```diff
@@ -193,18 +194,25 @@ const handleApprovedNameRequestRenew = (item: Business): void => {
   }
 }
 
-const handleApprovedNameRequestChangeName = (item: Business, nrRequestActionCd: NrRequestActionCodes): void => {
-  if (!isModernizedEntity(item)) {
+const handleApprovedNameRequestChangeName = async (item: Business, nrRequestActionCd: NrRequestActionCodes): Promise<void> => {
+  // Extract corporation number from the name request
+  const corpNum = item.nameRequest?.corpNum
+  if (!corpNum) { return } // Early return if no corpNum exists
+
+  // Show loading state while checking business existence
+  isButtonActionProcessing.value = true
```

Comment:
This is to prevent the user from pressing the button again and again.

Reply:
A spinner overlay could also be used.

Reply:
I tried using a spinner overlay but I didn't like how it fit in the code. 

What do you mean the drop-down menu is still active here?

Reply:
The drop-down menu is the small menu beside the main action button. While the main action button has a loading indicator, the drop-down menu (down arrow) appears to still be enabled -- though COLIN opens very quickly in a new tab and I'm unable to see if I can cause trouble by doing something in the dropdown menu while the LEAR lookup is happening. So it's mostly a UX thing I guess.

OK for the loading state vs spinner.